nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/layers/recurrent/LSTMLayer.java

```diff
@@ -114,31 +118,63 @@ public LSTMLayer(INDArray x, INDArray cLast, INDArray yLast, INDArray maxTSLengt
         throw new UnsupportedOperationException("Not yet implemented");
     }
 
-    @Override
-    public void initFromTensorFlow(NodeDef nodeDef, SameDiff initWith, Map<String, AttrValue> attributesForNode, GraphDef graph) {
-        configuration = LSTMConfiguration.builder()
-                .forgetBias(attributesForNode.get("forget_bias").getF())
-                .clippingCellValue(attributesForNode.get("cell_clip").getF())
-                .peepHole(attributesForNode.get("use_peephole").getB())
-                .dataFormat(RnnDataFormat.TNS)  //Always time major for TF BlockLSTM
-                .build();
-        addIArgument(configuration.iArgs(true));
-        addTArgument(configuration.tArgs());
-    }
 
     @Override
     public String opName() {
-        return "lstmBlock";
+        return "lstmLayer";
     }
 
     @Override
     public Map<String, Object> propertiesForFunction() {
-        return configuration.toProperties(true);
+        return configuration.toProperties(true, true);
     }
 
-    @Override
-    public String tensorflowName() {
-        return "BlockLSTM";
+
+    public long[] iArgs(LSTMLayerConfig configuration) {
+        return new long[]{
+                (long) configuration.toProperties(true, true).get("LSTMDataFormat"),  // INT_ARG(0)
+                (long) configuration.toProperties(true, true).get("LSTMDirectionMode") , // INT_ARG(1)
+                (long) configuration.toProperties(true, true).get("gateAct"),  // INT_ARG(2)
+                (long) configuration.toProperties(true, true).get("outAct") , // INT_ARG(3)
+                (long) configuration.toProperties(true, true).get("cellAct")  // INT_ARG(4)
+
+        };
     }
 
+    public double[] tArgs(LSTMLayerConfig configuration) {
+        return new double[]{(double) configuration.toProperties(true, true).get("cellClip")}; // T_ARG(0)
+    }
+
+
+    public boolean[] bArgs(LSTMLayerWeights weights, INDArray maxTSLength, INDArray yLast, INDArray cLast, LSTMLayerConfig configuration) {
+        return new boolean[]{
+                weights.hasBias(),         // hasBiases: B_ARG(0)
+                maxTSLength != null,         // hasSeqLen: B_ARG(1)
+                yLast != null,               // hasInitH: B_ARG(2)
+                cLast != null,              // hasInitC: B_ARG(3)
+                weights.hasPH(),          // hasPH: B_ARG(4)
+                (boolean) configuration.toProperties(true, true).get("retFullSequence"), //retFullSequence: B_ARG(5)
+                (boolean) configuration.toProperties(true, true).get("retLastH"),  //  retLastH: B_ARG(6)
+                (boolean) configuration.toProperties(true, true).get("retLastC")   // retLastC: B_ARG(7)
```

Comment:
As above - use methods like configuration.isRetFullSequence() and the like. Don't use configuration.toProperties approach


And yes actually (you asked about generics) - we should be able to do this and have the method work for both SDVariable and INDArray, instead of having 2 bArgs methods
```
public <T> boolean[] bArgs(LSTMLayerWeights weights, T maxTSLength, T yLast, T cLast, LSTMLayerConfig configuration) {
```